datashuttle/configs.py

```diff
@@ -0,0 +1,184 @@
+import copy
+import pathlib
+import warnings
+from collections import UserDict
+from pathlib import Path
+from typing import Any, Union
+
+import yaml
+
+from datashuttle.utils_mod import utils
+
+
+class Configs(UserDict):
+    """ """
+
+    def __init__(self, file_path, input_dict):
+        super(Configs, self).__init__(input_dict)
+
+        self.file_path = file_path
+
+    def check_dict_values_and_inform_user(self):
+        """
+        Check the values of the current dictionary are set
+        correctly and will not cause downstream errors.
+        """
+
+        # Check relevant remote_path is set
+        if self["ssh_to_remote"]:
+            if not self["remote_path_ssh"]:
+                utils.raise_error(
+                    "ssh to remote is on but remote_path_ssh "
+                    "has not been set."
+                )
+        else:
+            if not self["remote_path_local"]:
+                utils.raise_error(
+                    "ssh to remote is off but remote_path_local "
+                    "has not been set."
+                )
+
+        # Check bad remote path format
+        if self.get_remote_path().as_posix()[0] == "~":
+            utils.raise_error(
+                "remote_path must contain the full directory path "
+                "with no ~ syntax"
+            )
+
+        # Check SSH settings
+        if self["ssh_to_remote"] is True and (
+            not self["remote_host_id"] or not self["remote_host_username"]
+        ):
+            utils.raise_error(
+                "ssh to remote set but no remote_host_id or "
```

Comment:
seems like a typo. Should it be 'but no remote_host_id is set or no remote_host_username is provided'?